### DISCUSSION TITLE
Windows Development

### DISCUSSION THREAD

**Discussion by ajowsey** [nrobinson2000/neopo#19] [Q&A] [answered]
Hello @nrobinson2000 ,

I am finding particle workbench compile times painfully slow on my Windows development environment. I've been using neopo for over a year in my bitbucket pipeline (and love it) but would like to use it for my local development now as well because I read [here ](https://community.particle.io/t/how-much-faster-is-compiling-on-linux/57156/2) that my compile times could be halved.  

What would be the easiest way to incorporate neopo? Should I use wsl / windows docker? Or am I better off adding a full linux work environment to my life?

**Comment by nrobinson2000** [comment, marked as answer]
Hey @ajowsey 

Thanks for using neopo!

Neopo should work well in WSL. The installation process in WSL is essentially the same as in a regular Linux environment.

neopo will work even better in a native Linux environment, if that is something you're interested in.

Personally, I've been using Arch Linux on virtually all of my Linux machines, and I provide a PKGBUILD for cleanly installing neopo on Arch-based distros: https://aur.archlinux.org/packages/neopo-git

I've kind of taken a break from neopo and Particle, but I'm open to suggestions for fixes and improvements to neopo.

**Reply by ajowsey** [reply]
Using WSL, would I still able to do USB flash? I haven't worked with it before...

**Reply by ajowsey** [reply]
Hey Nathan, I think will use WSL in combination with https://github.com/particle-iot/device-os-flash-util
